### PR TITLE
[ISSUE #3343]Optimize Codec.Encoder and Codec.Decoder

### LINKED ISSUE #3343: [Enhancement] Optimize Codec.Encoder and Codec.Decoder
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image
image
optimize Decoder and Encoder logic, remove add 8

### Describe the solution you'd like

optimize Decoder and Encoder logic, remove add 8.
and add package diagram of the data structure

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by mxsm**
Fixes #3343 

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications

*Optimize Codec.Encoder and Codec.Decoder*

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by xwm1992** [comment]
@lrhkobe please help review this pr.

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/protocol/tcp/codec/Codec.java [hunk 1]
```diff
@@ -74,12 +74,20 @@ public void encode(ChannelHandlerContext ctx, Package pkg, ByteBuf out) throws E
             int headerLength = ArrayUtils.getLength(headerData);
             int bodyLength = ArrayUtils.getLength(bodyData);
 
-            int length = 4 + 4 + headerLength + bodyLength;
+            int length = headerLength + bodyLength;
```
Comment (xwm1992) [inline eventmesh-common/src/main/java/org/apache/eventmesh/common/protocol/tcp/codec/Codec.java:77]: this line need to change:
`int length = CONSTANT_MAGIC_FLAG.length + VERSION.length + headerLength + bodyLength;`

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/protocol/tcp/codec/Codec.java [hunk 2]
```diff
@@ -108,14 +116,14 @@ public void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) thro
 
                 final int length = in.readInt();
                 final int headerLength = in.readInt();
-                final int bodyLength = length - 8 - headerLength;
+                final int bodyLength = length - headerLength;
```
Comment (xwm1992) [inline eventmesh-common/src/main/java/org/apache/eventmesh/common/protocol/tcp/codec/Codec.java:119]: this line need to change:
`final int bodyLength = length - CONSTANT_MAGIC_FLAG.length - VERSION.length - headerLength;`

**Review by xwm1992 — CHANGES_REQUESTED** [review]

**Comment by mxsm** [comment]
@xwm1992  done, PTAL~

**Pull request merged**